src/requirements/WalletActivity/components/BlockNumberFormControl.tsx

```diff
@@ -75,6 +79,7 @@ const BlockNumberFormControl = ({
     register(`${baseFieldPath}.data.${dataFieldName}`, {
       required:
         !!isRequired && !!shouldFetchBlockNumber && "This field is required.",
+      validate: () => (isInvalid ? invalidText : undefined),
```

Comment:
I think we could use the native HTML `min` & `max` attributes on the date pickers, and just make sure that the user can't pick invalid dates. You can see an example inside the `EditRewardAvailabilityModal` component.